spec/features/reservations_spec.rb

```diff
@@ -376,4 +414,214 @@
       end
     end
   end
+
+  context 'valid items on confirmation page' do
+    before(:each) do
+      empty_cart
+      select('10', from: 'items_per_page')
+      find('#items_per_form').submit_form!
+      add_item_to_cart(@eq_model)
+      add_item_to_cart(@eq_model2)
+      update_cart_start_date(Time.zone.today)
+      due_date = Time.zone.today + 1.day
+      update_cart_due_date(due_date)
+      visit new_reservation_path
+    end
+
+    shared_examples 'can make valid change to reservation' do |reserver|
+      let(:reserver) { reserver }
+      str = reserver ? 'for other user successfully' : 'successfully'
+      it str do
+        quantity_forms = page.all('#quantity_form')
+        fill_in "quantity_field_#{@eq_model.id}", # edit and submit
+                with: (@eq_model.max_per_user)
+        quantity_forms[0].submit_form!
+        expect(page).to have_content 'Confirm Reservation'
```

Comment:
I think you need to reload the page here to verify that everything still works. Also, it doesn't look like you're testing to make sure that your change to the cart was actually saved. We could add another expectation here for that.